libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/idp/SPRequestHandler.java

```diff
@@ -26,10 +26,131 @@
  */
 package com.salesforce.androidsdk.auth.idp;
 
+import android.app.Activity;
+import android.content.Intent;
+import android.os.AsyncTask;
+
+import com.salesforce.androidsdk.app.SalesforceSDKManager;
+import com.salesforce.androidsdk.auth.HttpAccess;
+import com.salesforce.androidsdk.auth.OAuth2;
+import com.salesforce.androidsdk.config.BootConfig;
+import com.salesforce.androidsdk.security.SalesforceKeyGenerator;
+import com.salesforce.androidsdk.util.SalesforceSDKLogger;
+
+import java.net.URI;
+
 /**
  * This class performs requests made by the SP app based on the IDP app's response.
  *
  * @author bhariharan
  */
 public class SPRequestHandler {
+
+    public static final int IDP_REQUEST_CODE = 375;
+    private static final String CODE_VERIFIER = "code_verifier";
+    private static final String TAG = "SPRequestHandler";
+
+    private String codeVerifier;
+    private String codeChallenge;
+    private String loginUrl;
+    private SPConfig spConfig;
+
+    /**
+     * Parameterized constructor.
+     *
+     * @param loginUrl Login URL.
+     */
+    public SPRequestHandler(String loginUrl) {
+        codeVerifier = CODE_VERIFIER;
```

Comment:
Running into some minor issues with `Base64` encoding. Will look into it after merging this PR. Using a random string works for now.